src/contracts/interfaces/IStrategyManager.sol

```diff
@@ -13,6 +13,29 @@ import "./IEigenPodManager.sol";
  * @notice See the `StrategyManager` contract itself for implementation details.
  */
 interface IStrategyManager {
+    /// @dev Thrown when deposit amount would exceed max.
+    error DepositWouldExceedMax();
```

Comment:
This is actually not for exceeding a deposit amount (although there is a `MAX_TOTAL_SHARES` defined in the StrategyBase contract). But we limit the number of deposited strategies a staker can deposit into because of gas limitations when attempting to delegate/undelegate because it reads and writes to storage for ALL your currently deposited strategies. 
Actually because there are increased gas costs from the Slashing release, we will need to account for increased gas costs and see if this arbitrary 32 constant is still feasible.